typing/primitive.ml

```diff
@@ -330,38 +332,54 @@ let native_name p =
 let byte_name p =
   p.prim_name
 
-let equal_boxed_vector v1 v2 =
-  match v1, v2 with
-  | Pvec128, Pvec128 -> true
+let unbox_integer = function
+  | Boxed_int32 -> Unboxed_int32
+  | Boxed_nativeint -> Unboxed_nativeint
+  | Boxed_int64 -> Unboxed_int64
+
+let unbox_float = function
+  | Boxed_float32 -> Unboxed_float32
+  | Boxed_float64 -> Unboxed_float64
+
+let unbox_vector = function
+  | Boxed_vec128 -> Unboxed_vec128
+
+(* since these are just constant constructors, we can just use polymorphic
+   equality and comparison at no performance loss: *)
+let equal_unboxed_integer
+      ((Unboxed_int32 | Unboxed_nativeint | Unboxed_int64) as i1) i2 = i1 = i2
```

Comment:
Or just: equal_unboxed_integer (i1 : unboxed_integer) i2 = i1 = i2?

Reply:
The match is so that you can verify that it's only constant constructors, and therefore polymorphic equality is appropriate